MarkupViewer.py

```diff
@@ -850,15 +847,45 @@ def __init__(self, parent=None):
             for b in (spacer, self.download, self.go_to_gh, self.cancel):
                 tb.addWidget(b)
 
-            self.addToolBar(0x8, tb)
+            self.addToolBar(0x8, tb) 
+
+def error(msg):
+    print(msg)
+    exit()
+
+class GeoAction(argparse.Action):
+    def __init__(self, option_strings, dest, nargs=None, **kwargs):
+        if nargs:
+            error("Too many arguments for --geometry")
+
+        super().__init__(option_strings, dest, **kwargs)
+
+    def __call__(self, parser, namespace, value, option_string=None):
+        try:
+            values = value.split("x")
+            setattr(namespace, self.dest, (int(values[0]), int(values[1])))
+        except:
+            error("Invalid option " + value + " for --geometry!")
+
+
+class FmAction(argparse.Action):
+    def __init__(self, option_strings, dest, nargs=None, **kwargs):
+
+        super().__init__(option_strings, dest, **kwargs)
```

Comment:
Same as above.

Also it is complaining about `description` argument, I’m not sure why
```
Traceback (most recent call last):
  File ".\test.py", line 898, in <module>
    main()
  File ".\test.py", line 886, in main
    parser.add_argument('infile', nargs='?', default='', action=FmAction, description="Name of the markup file to be previewed.")
  File "C:\Python27\lib\argparse.py", line 1294, in add_argument
    action = action_class(**kwargs)
    super(FmAction, self).__init__(option_strings, dest, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'description'
```